catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchFilterVisitor.java

```diff
@@ -271,16 +269,20 @@ public Object visit(PropertyIsEqualTo filter, Object data) {
     }
 
     if (openSearchFilterVisitorObject.getCurrentNest() != NestedTypes.NOT) {
-      if (filter instanceof IsEqualsToImpl) {
-        IsEqualsToImpl isEqualsTo = (IsEqualsToImpl) filter;
-        Expression leftValue = isEqualsTo.getLeftValue();
-        if (Metacard.ID.equals(leftValue.toString())) {
-          openSearchFilterVisitorObject.setId(isEqualsTo.getExpression2().toString());
-        }
-      } else if (filter instanceof PropertyIsEqualToLiteral) {
-        PropertyIsEqualToLiteral isEqualsTo = (PropertyIsEqualToLiteral) filter;
-        if (Metacard.ID.equals(isEqualsTo.getExpression1().toString())) {
-          openSearchFilterVisitorObject.setId(isEqualsTo.getExpression2().toString());
+      final Expression expression1 = filter.getExpression1();
+      if (expression1 instanceof PropertyName) {
+        final String propertyName = ((PropertyName) expression1).getPropertyName();
+        final String expectedPropertyIsEqualToTerm = Metacard.ID;
+        if (StringUtils.equals(expectedPropertyIsEqualToTerm, propertyName)) {
```

Comment:
Is there a reason we don't just do `expectedPropertyIsEqualToTerm.equals(propertyName)`?